lib/logstash/outputs/file.rb

```diff
@@ -202,25 +204,23 @@ def extract_file_root
 
   private
   def flush(fd)
-    if flush_interval > 0
-      flush_pending_files
-    else
-      fd.flush
-    end
+    fd.flush
   end
 
-  # every flush_interval seconds or so (triggered by events, but if there are no events there's no point flushing files anyway)
+  # the back-bone of @flusher, our periodic-flushing interval.
   private
   def flush_pending_files
-    return unless Time.now - @last_flush_cycle >= flush_interval
-    @logger.debug("Starting flush cycle")
+    @io_mutex.synchronize do
+      @logger.debug("Starting flush cycle")
 
-    @files.each do |path, fd|
-      @logger.debug("Flushing file", :path => path, :fd => fd)
-      fd.flush
+      @files.each do |path, fd|
+        @logger.debug("Flushing file", :path => path, :fd => fd)
+        fd.flush
+      end
     end
-
-    @last_flush_cycle = Time.now
+  rescue => e
+    # squash exceptions caught while flushing after logging them
+    @logger.error("Exception flushing files", exception: e.message, backtrace: e.backtrace)
```

Comment:
minor style notice: let's use `=>` syntax here for the k/v pairs, as we do in most of the logstash/plugins code (e.g. line 217)